reproducibility/codalab/util/analysis_utils.py

```diff
@@ -0,0 +1,120 @@
+                # message if the file does not exist.
+                try:
+                    id_log_file = f"id_{split}_{log_type}"
+                    result_dfs[id_log_file] = pd.read_csv(
+                        os.path.join(log_url, f"{id_log_file}.csv")
+                    )
+                except:
+                    print(f"{id_log_file} not available at {log_url}")
+
+            result_dfs[f"{split}_{log_type}"] = pd.read_csv(log_csv_path)
+    return result_dfs
+
+
+def compile_results(dataset, results, in_distribution_val=False):
+    metrics = get_metrics(dataset)
+    compiled_results = dict()
+
+    for algorithm, result_dfs in results.items():
+        compiled_results[algorithm] = dict()
+        print(f"\ndataset={dataset}, algorithm={algorithm}:")
+
+        for result_type in RESULT_TYPES:
+            if len(result_dfs) > 0 and result_type in result_dfs[0]:
+                compiled_results[algorithm][result_type] = dict()
+                for metric in metrics:
+                    compiled_results[algorithm][result_type][metric] = []
+
+        for result_df in result_dfs:
+            for result_type in RESULT_TYPES:
+                if result_type not in result_df:
+                    continue
+
+                for metric in metrics:
+                    sort_metrics = [metric]
+                    # Use the "acc_avg" metric as a tie-breaker for datasets that have "acc_avg"
+                    if metric != "acc_avg" and dataset in [
+                        "amazon",
+                        "camelyon17",
+                        "fmow",
+                        "iwildcam",
+                    ]:
+                        sort_metrics.append("acc_avg")
+
+                    val_split = "id_val_eval" if in_distribution_val else "val_eval"
+                    result = get_early_stopped_row(
+                        result_df[result_type],
+                        result_df[val_split],
+                        sort_fields=sort_metrics,
+                    )
+                    compiled_results[algorithm][result_type][metric].append(
+                        result[metric]
+                    )
+
+        for result_type in RESULT_TYPES:
+            if result_type in compiled_results[algorithm]:
+                for metric in metrics:
+                    # Take the standard deviation and mean across replicants for each metric
+                    compiled_results[algorithm][result_type][f"{metric}_std"] = np.std(
+                        compiled_results[algorithm][result_type][metric]
+                    )
+                    compiled_results[algorithm][result_type][metric] = np.mean(
+                        compiled_results[algorithm][result_type][metric]
+                    )
+    return compiled_results
+
+
+def get_early_stopped_row(src_df, val_df, sort_fields, ascending=False):
+    epoch = val_df.sort_values(sort_fields, ascending=ascending).iloc[0]["epoch"]
```

Comment:
Just to double check, all of the datasets have evals such that higher is better, right? (The only one that's a bit wonky is povertymap which uses MSE for val, but the eval should be Pearson R?)

Reply:
I actually never called this function with `ascending=True`, so I will just remove the parameter.